File: be_camphub/src/main/java/org/camphub/be_camphub/entity/ReturnRequest.java
package org.camphub.be_camphub.entity;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import jakarta.persistence.*;

import org.camphub.be_camphub.enums.ReasonReturnType;
import org.camphub.be_camphub.enums.ReturnRequestStatus;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Table(name = "return_requests")
@FieldDefaults(level = AccessLevel.PRIVATE)
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ReturnRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    @Column(nullable = false)
    UUID bookingId;

    @Column(nullable = false)
    UUID lesseeId;

    @Column(nullable = false)
    UUID lessorId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    ReasonReturnType reason;

    @ElementCollection
    @CollectionTable(name = "return_request_evidences", joinColumns = @JoinColumn(name = "return_request_id"))
    @Column(name = "evidence_urls")
    List<MediaResource> evidenceUrls;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    ReturnRequestStatus status;

    @Column(length = 500)
    String note;

    @Column(length = 500)
    String adminNote;

    @CreatedDate
    @Column(updatable = false)
    LocalDateTime createdAt;

    @LastModifiedDate
    LocalDateTime updatedAt;

    LocalDateTime resolvedAt;
}

File: fe_camphub/src/libs/types/ReturnRequest.ts
import { ReasonReturnType, ReturnRequestStatus } from "../constants";
import { MediaResource } from "./MediaResource";

export interface ReturnRequest {
    id: string;
    bookingId: string;
    lesseeId: string;
    lessorId: string;
    reason: ReasonReturnType;       // lý do trả đồ
    evidenceUrls?: MediaResource[]; // minh chứng
    status: ReturnRequestStatus;
    note?: string | null;
    adminNote?: string | null;
    createdAt: string;
    updatedAt: string;
    resolvedAt?: string | null;
}
